src/modules/cmdpal/Microsoft.CmdPal.UI/App.xaml.cs

```diff
@@ -77,6 +85,11 @@ private static ServiceProvider ConfigureServices()
         services.AddSingleton<ICommandProvider, RegistryCommandsProvider>();
         services.AddSingleton<ICommandProvider, WindowsSettingsCommandsProvider>();
 
+        // Models
+        services.AddSingleton<TopLevelCommandManager>();
+        services.AddSingleton<MainListPage>();
```

Comment:
We shouldn't have pages here, nor should we handle/reference them ever in the ViewModel where you retrieved this in the ShellViewModel.

The MainListPage has access to the Services bucket already, as it can go `App.Current.Services` exposed on L33/42 above.

Reply:
Oh even for this page which is an `IListPage` model, not a View?

Reply:
Yeah, this is a weird case, I forgot. It'd still be odd to have a model item here. The main thing we'll use the services bucket for is the services and the ViewModels which rely on them. ViewModels should be the mediator between the model entities and the View itself in our XAML.